client/my-sites/stats/stats-module/all-time-nav.jsx

```diff
@@ -69,20 +70,30 @@ export const StatsModuleSummaryLinks = ( props ) => {
 		recordStats( item );
 	};
 
-	// Template for standard range options (7, 30, Quarter, Year, All Time).
-	const summaryPath = `/stats/day/${ path }/${ siteSlug }?startDate=${ moment().format(
-		'YYYY-MM-DD'
-	) }&summarize=1&num=`;
+	const summaryPeriodPath = useMemo( () => {
+		let periodPath = `/stats/${
+			period.period
+		}/${ path }/${ siteSlug }?startDate=${ period.endOf.format( 'YYYY-MM-DD' ) }`;
+
+		// Override if custom range was used in query.
+		if ( query.start_date ) {
+			periodPath = `/stats/${ period.period }/${ path }/${ siteSlug }?startDate=${ query.start_date }&endDate=${ query.date }`;
+		}
+		return periodPath;
+	}, [ period.period, path, siteSlug, query ] );
 
-	// Path for summary or custom range option. ie: The first button in the row.
-	// Defaults to one day/week/month/year.
-	let summaryPeriodPath = `/stats/${
-		period.period
-	}/${ path }/${ siteSlug }?startDate=${ period.endOf.format( 'YYYY-MM-DD' ) }`;
-	// Override if custom range was used in query.
-	if ( query.start_date ) {
-		summaryPeriodPath = `/stats/${ period.period }/${ path }/${ siteSlug }?startDate=${ query.start_date }&endDate=${ query.date }`;
-	}
+	const getSummaryPathForDaysRange = useMemo( () => {
+		return ( numberDays ) => {
+			const queryParams = new URLSearchParams( context.query );
+
+			queryParams.set( 'startDate', moment().format( 'YYYY-MM-DD' ) );
+			queryParams.set( 'summarize', 1 );
+			queryParams.set( 'num', numberDays );
+			queryParams.delete( 'endDate' );
+
+			return `/stats/day/${ path }/${ siteSlug }?${ queryParams.toString() }`;
+		};
+	}, [ path, siteSlug, query, context.query ] );
```

Comment:
```suggestion
	}, [ path, siteSlug, context.query ] );
```
<img width="879" alt="截圖 2025-04-22 凌晨12 01 30" src="https://github.com/user-attachments/assets/f06a37b7-8945-44ee-8356-433ca24814d0" />

Reply:
This is fixed and I now have eslint setup in the IDE so these will be caught sooner